agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/connectionstring/ConnectionString.java

```diff
@@ -0,0 +1,192 @@
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.azure.monitor.opentelemetry.exporter.implementation.connectionstring;
+
+import com.azure.core.util.CoreUtils;
+import com.google.auto.value.AutoValue;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class ConnectionString {
+
+  // visible for testing
+  static final int CONNECTION_STRING_MAX_LENGTH = 4096;
+
+  public static ConnectionString parse(String connectionString) {
+    return parse(connectionString, null, null);
+  }
+
+  public static ConnectionString parse(
+      String connectionString,
+      @Nullable String statsbeatInstrumentationKey,
+      @Nullable String statsbeatIngestionEndpoint) {
+    return mapToConnectionConfiguration(
+        getKeyValuePairs(connectionString),
+        statsbeatInstrumentationKey,
+        statsbeatIngestionEndpoint);
+  }
+
+  public abstract String getInstrumentationKey();
+
+  public abstract URL getIngestionEndpoint();
+
+  public abstract URL getLiveEndpoint();
+
+  public abstract URL getProfilerEndpoint();
+
+  public abstract String getStatsbeatInstrumentationKey();
+
+  public abstract URL getStatsbeatEndpoint();
+
+  private static Map<String, String> getKeyValuePairs(String connectionString) {
+    if (connectionString.length() > CONNECTION_STRING_MAX_LENGTH) { // guard against malicious input
+      throw new IllegalArgumentException(
+          "ConnectionString values with more than "
+              + CONNECTION_STRING_MAX_LENGTH
+              + " characters are not allowed.");
+    }
+    // parse key value pairs
+    Map<String, String> kvps;
+    try {
+      kvps = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+      kvps.putAll(extractKeyValuesFromConnectionString(connectionString));
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("Could not parse connection string.", e);
+    }
+
+    return kvps;
+  }
+
+  private static Map<String, String> extractKeyValuesFromConnectionString(String connectionString) {
+    Objects.requireNonNull(connectionString);
+    Map<String, String> keyValues = new HashMap<>();
+    String[] splits = connectionString.split(";");
+    for (String split : splits) {
+      String[] keyValPair = split.split("=");
+      if (keyValPair.length == 2) {
+        keyValues.put(keyValPair[0], keyValPair[1]);
+      }
+    }
+    return keyValues;
+  }
+
+  private static ConnectionString mapToConnectionConfiguration(
```

Comment:
we can call ConnectionString.parse(statsbeat ikey, statsbeat endpoint) separately.

Reply:
or ConnectionString.parseStatsbeat?